### ISSUE TITLE
cpp-netlib 0.12.0-final does not compile with C++17 gcc + libstdc++

### ISSUE THREAD

**Issue by germandiago** [cpp-netlib/cpp-netlib#863]
Hello everyone, 

OS: ubuntu 18.04 (dockerized)

There seems to be an ambiguity in `make_optional` calls due to ADL when compiling with gcc + libstdc++ as packaged in Ubuntu. For some reason std::make_optional is included indirectly in file `uri_parts.hpp` file. I tried to include cppnet headers before anything else but the problem persisted.

So I created a patch for explicit ::boost qualification in `uri_parts.hpp`. If there is any interest, please let me know, I will send a PR.

**Comment by deanberris** [comment]
Yes, please send a PR -- you can assign it to me or to @glynos for review/merging.
